tests/common/include/tests/common/energy_loss.inl

```diff
@@ -158,38 +167,155 @@ TEST_P(EnergyLossLandauValidation, landau_energy_loss) {
+    const scalar Landau_MeV{
+        I.compute_energy_loss_landau(is, slab, pdg, m, qOverP, q) /
         unit<scalar>::MeV};
 
     // Check if difference is within 5% error
-    EXPECT_TRUE(std::abs(std::get<2>(GetParam()) - dE) / dE < 0.05f);
+    EXPECT_TRUE(std::abs(std::get<2>(GetParam()) - Landau_MeV) / Landau_MeV <
+                0.05f);
 
     // Landau Energy loss Fluctuation
-    const scalar fwhm{
-        I.compute_energy_loss_landau_fwhm(is, slab, pdg, m, qOverP, -1.f) /
+    const scalar fwhm_MeV{
+        I.compute_energy_loss_landau_fwhm(is, slab, pdg, m, qOverP, q) /
         unit<scalar>::MeV};
 
     // Check if difference is within 10% error
-    EXPECT_TRUE(std::abs(std::get<3>(GetParam()) - fwhm) / fwhm < 0.1f);
+    EXPECT_TRUE(std::abs(std::get<3>(GetParam()) - fwhm_MeV) / fwhm_MeV < 0.1f);
 }
 
 // Expected output from Fig 33.7 in RPP2018
 INSTANTIATE_TEST_SUITE_P(Landau_10GeV_Silicon, EnergyLossLandauValidation,
                          ::testing::Values(std::make_tuple(
                              silicon<scalar>(), 10.f * unit<scalar>::GeV,
                              0.525f, 0.13f)));
+
+// Input tuple: < energy >
+class LandauDistributionValidation
+    : public ::testing::TestWithParam<std::tuple<scalar>> {};
+
+TEST_P(LandauDistributionValidation, landau_distribution) {
+
+    // Random generator
+    std::random_device rd{};
+    std::mt19937_64 generator{rd()};
+    generator.seed(0u);
+
+    // Interaction object
+    interaction<scalar> I;
+
+    // intersection with a zero incidence angle
+    intersection2D<sf_handle_t> is;
+    is.cos_incidence_angle = 1.f;
+
+    // Material
+    const auto mat = silicon<scalar>();
+
+    // Thickness
+    const auto thickness = 1.f * unit<scalar>::mm;
+
+    // Material slab with a unit thickness
+    material_slab<scalar> slab(mat, thickness);
+
+    // muon
+    constexpr int pdg{pdg_particle::eMuon};
+
+    // mass
+    constexpr scalar m{105.7f * unit<scalar>::MeV};
+
+    // Energy
+    const scalar E = std::get<0>(GetParam());
+
+    // p
+    const scalar p = std::sqrt(E * E - m * m);
+
+    // q
+    const scalar q = -1.f;
+
+    // qOverP
+    const scalar qOverP{q / p};
+
+    // Bethe energy loss
+    const scalar dE{I.compute_energy_loss_bethe(is, slab, pdg, m, qOverP, q)};
+
+    // Landau Energy loss
+    const scalar mpv{I.compute_energy_loss_landau(is, slab, pdg, m, qOverP, q)};
+
+    // Landau Energy loss Sigma
+    const scalar sigma{
+        I.compute_energy_loss_landau_sigma(is, slab, pdg, m, qOverP, q)};
+
+    // Landau Sampling
+    landau_distribution<scalar> landau;
+
+    std::vector<scalar> samples;
+    std::size_t n_samples{10000u};
+    for (std::size_t i = 0u; i < n_samples; i++) {
+        const scalar sa = landau(generator, mpv, sigma);
+        samples.push_back(sa);
+    }
+
+    // Mean energy loss from Landau samples
+    const scalar sample_mean = statistics::mean(samples);
```

Comment:
If you get ```std::numeric_limits<scalar>::max()``` from the landau distribution, should these go into the mean?

Reply:
Technically, it won't produce `std::numeric_limits<scalar>::max()` because we put `z` in [0,1] range